main.go

```diff
@@ -30,13 +31,15 @@ func main() {
 	log.SetFormatter(new(eventlogger.CustomFormatter))
 	log.SetLevel(getLogLevel())
 
+	httpClient := &http.Client{}
```

Comment:
I'm not really sure what this is, but I see that we are passing it around quite a lot.

Reply:
Before, we were creating an http client every time we were sending a callback, but I think we should create it once and [reuse it everywhere](https://pkg.go.dev/net/http#:~:text=Clients%20and%20Transports%20are%20safe%20for%20concurrent%20use%20by%20multiple%20goroutines%20and%20for%20efficiency%20should%20only%20be%20created%20once%20and%20re-used.)